worker/uniter/runner/jujuc/context.go

```diff
@@ -95,6 +96,19 @@ type actionHookContext interface {
 	LogActionMessage(string) error
 }
 
+// unitCacheContext is cache for charm state to be held in the context.
+type unitCacheContext interface {
```

Comment:
If we want to break from the everything-is-some-kind-of-context pattern which is everywhere in the uniter we could rename this to something like `unitCacheReadWriter`. Guess since all other interfaces names are following the `xyzContext` pattern we are probably fine.